mobile/lib/pages/editing/crop.page.dart

```diff
@@ -0,0 +1,185 @@
+import 'package:flutter/material.dart';
+import 'package:crop_image/crop_image.dart';
+import 'edit.page.dart';
+
+class CropImagePage extends StatefulWidget {
+  final Image image;
+  const CropImagePage({super.key, required this.image});
+  @override
+  State<CropImagePage> createState() => _CropImagePageState();
+
+}
+class _CropImagePageState extends State<CropImagePage> {
+  late CropController _cropController;
```

Comment:
In general, we prefer using `HookWidget` from [flutter_hooks](https://pub.dev/packages/flutter_hooks) to handle widget state instead of using `setState`. You would use something like:

```dart
class CropImagePage extends HookWidget {
  @override
  Widget build(BuildContext context) {
    final cropController = useCropController();
    final aspectRatio = useState<double?>(null);
}
```

You may have to create your own `useCropController` function:

```dart
CropController useCropController() {
  return use(CropController(defaultCrop: ...));
}
```

Reply:
After a serious crisis, I still could not figure out how I could use the HookWidget on the cropscreen. Could you help me on this?

Reply:
Sure! You'd change this to be a HookWidget:

```dart
class CropImage extends HookWidget {
```

Then, instead of any of your stateful member variables, use Hooks instead!

```dart
@override

Widget build(BuildContext context) {
  final cropController = useCropController();
  final aspectRatio = useState<double?>(null);

  ...
}
```

And then, whenever you'd normally reference the `aspectRatio`, instead reference the `aspectRatio.value` instead.

```dart
aspectRatio.value == 16.0 / 9.0;
```

You will then need to create your own `useCropController` function (as above). You can put the file somewhere in `lib/utils/hooks/crop_controller_hook.dart`.

Finally, you will no longer need to use `setState` since the hooks will take care of the state.

Reply:
Done!!